### PR TITLE
media-sound/cantata: version bump

### PR DESCRIPTION
**Pull request by Civil**
Package-Manager: portage-2.2.28

Fixes: https://bugs.gentoo.org/show_bug.cgi?id=544734

### CODE REVIEW

**Comment by Arfrever** [comment]
`sys-devel/gettext` is needed only when support for KDE is enabled.
See: https://github.com/CDrummond/cantata/blob/v2.0.0/po/CMakeLists.txt#L17
So `kde? ( sys-devel/gettext )` should be used in DEPEND.

`QT_SELECT` variable must be exported, because otherwise either:
- `lconvert` and `lrelease` from wrong version of Qt can be used.
- `lconvert` and `lrelease` will fail with errors.

Default version of Qt for usage of tools such as `lconvert`, `lrelease`, `moc`, `qmake` outside of building controlled by package manager is specified in `/etc/xdg/qtchooser/default.conf`.
Currently `/etc/xdg/qtchooser/default.conf` is installed only by `dev-qt/qtcore:4` and it contains:
```
/usr/lib64/qt4/bin
/usr/lib64/qt4
```
If `dev-qt/qtcore:4` is installed and `QT_SELECT` is not exported, then `lconvert`, `lrelease` from Qt 4 would be used even when `USE="qt5"`.

If `dev-qt/qtcore:4` is not installed and `QT_SELECT` is not exported, then the following build failure will occur:
```
Scanning dependencies of target translations
make[2]: Leaving directory '/var/tmp/portage/media-sound/cantata-2.0.0/work/cantata-2.0.0_build'
make -f po/CMakeFiles/translations.dir/build.make po/CMakeFiles/translations.dir/build
make[2]: Entering directory '/var/tmp/portage/media-sound/cantata-2.0.0/work/cantata-2.0.0_build'
cd /var/tmp/portage/media-sound/cantata-2.0.0/work/cantata-2.0.0/po && /usr/bin/lconvert ru.po -o /var/tmp/portage/media-sound/cantata-2.0.0/work/cantata-2.0.0_build/po/cantata_ru.ts -of ts
lconvert: could not find a Qt installation of ''
po/CMakeFiles/translations.dir/build.make:57: recipe for target 'translations' failed
```
Solution:
```
pkg_setup() {
	if use qt5; then
		export QT_SELECT=5
	else
		export QT_SELECT=4
	fi
}
```

**Comment by monsieurp** [comment]
@gentoo/kde

**Comment by monsieurp** [comment]
@Arfrever tattle tale

**Comment by kensington** [comment]
``QT_SELECT`` should not be relied upon - please use ``qt4_get_bindir`` and ``qt5_get_bindir`` from ``qmake-utils.eclass`` (I guess the source will need to be patched in this case).

**Comment by ff2000** [comment]
As stated in the bug report cantata also works with breeze, and I'm sure it will work with any freedesktop-compliant theme (The MESSAGE(...) is for builds on non-linux-systems). So could the hard dep on oxygen-icons be removed, or replaced with something that works for everyone?

**Comment by monsieurp** [comment]
@kensington ping

**Comment by Arfrever** [comment]
Ebuild using `qt4_get_bindir` and `qt5_get_bindir` as suggested by kensington and with other fixes:
https://home.apache.org/~arfrever/cantata-2.0.0.ebuild

**Comment by johu** [comment]
@Civil Could you please update the PR?

**Comment by Civil** [comment]
Ok, I'll do that later this week.

**Comment by Civil** [comment]
I've updated the PR using @Arfrever ebuild.

File: media-sound/cantata/cantata-2.0.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,146 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+KDE_REQUIRED="optional"
```
Comment (johu) [inline media-sound/cantata/cantata-2.0.0.ebuild]: This var can be dropped (left over ffrom kde4 eclass)

File: media-sound/cantata/cantata-2.0.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	musicbrainz? ( cdda taglib )
+	qt5? ( !kde )
+	replaygain? ( taglib )
+"
+
+RDEPEND="
+	sys-libs/zlib
+	x11-libs/libX11
+	dev-db/sqlite:3
+	kde-frameworks/oxygen-icons
```
Comment (johu) [inline media-sound/cantata/cantata-2.0.0.ebuild]: Please add an alternative with kde-frameworks/breeze-icons (where breeze icons should be the first choice as oxygen-icons are deprecated).

**Comment by johu** [comment]
@Civil Thanks for the update.  There are only 2 minor things open until we can pick this :)

**Comment by Civil** [comment]
Updated PR again.

**Comment by johu** [comment]
Thanks, picked as 06f9b9bb9f1ef48268fe62845c5dd8816cb3b58f.

**Pull request closed without merging**